tests/prime-bsi-express-one-video-stream.cpp

```diff
@@ -0,0 +1,164 @@
+#define EXPECT(e, ...)                                                         \
+    do {                                                                       \
+        if (!(e)) {                                                            \
+            char buf[1 << 8] = { 0 };                                          \
+            LOGE(__VA_ARGS__);                                                 \
+            snprintf(buf, sizeof(buf) - 1, __VA_ARGS__);                       \
+            throw std::runtime_error(buf);                                     \
+        }                                                                      \
+    } while (0)
+#define CHECK(e) EXPECT(e, "Expression evaluated as false: %s", #e)
+#define DEVOK(e) CHECK(Device_Ok == (e))
+#define OK(e) CHECK(AcquireStatus_Ok == (e))
+
+void
+reporter(int is_error,
+         const char* file,
+         int line,
+         const char* function,
+         const char* msg)
+{
+    auto stream = is_error ? stderr : stdout;
+    fprintf(stream,
+            "%s%s(%d) - %s: %s\n",
+            is_error ? "ERROR " : "",
+            file,
+            line,
+            function,
+            msg);
+    fflush(stream);
+}
+
+void
+setup(AcquireRuntime* runtime)
+{
+    CHECK(runtime);
+
+    auto dm = acquire_device_manager(runtime);
+    CHECK(dm);
+
+    AcquireProperties props = {};
+    OK(acquire_get_configuration(runtime, &props));
+
+    DEVOK(device_manager_select(dm,
+                                DeviceKind_Camera,
+                                SIZED(".*BSI.*") - 1,
+                                &props.video[0].camera.identifier));
+    DEVOK(device_manager_select(dm,
+                                DeviceKind_Storage,
+                                SIZED("tiff") - 1,
+                                &props.video[0].storage.identifier));
+
+    storage_properties_init(&props.video[0].storage.settings,
+                            0,
+                            SIZED(TEST ".tif"),
+                            nullptr,
+                            0,
+                            { .x = 1, .y = 1 });
+
+    props.video[0].camera.settings.binning = 1;
+    props.video[0].camera.settings.pixel_type = SampleType_u16;
+    props.video[0].camera.settings.shape = { .x = 2048, .y = 2048 };
+    props.video[0].camera.settings.exposure_time_us = 1e4; // 10 ms
+    props.video[0].max_frame_count = 100;
+
+    OK(acquire_configure(runtime, &props));
+}
+
+void
+acquire(AcquireRuntime* runtime)
+{
+    AcquireProperties props = {};
+    OK(acquire_get_configuration(runtime, &props));
+
+    const auto next = [](VideoFrame* cur) -> VideoFrame* {
+        return (VideoFrame*)(((uint8_t*)cur) + cur->bytes_of_frame);
+    };
+
+    const auto consumed_bytes = [](const VideoFrame* const cur,
+                                   const VideoFrame* const end) -> size_t {
+        return (uint8_t*)end - (uint8_t*)cur;
+    };
+
+    clock clock_{};
```

Comment:
Given the definitions in platform.h, I think the type needs the explicit struct.

```suggestion
    struct clock clock_{};
```

Maybe some preprocessors/compilers can infer that, but my version of Apple clang (14.0.3) could not, so I was unable to build before doing this.